src/Rules/PhpDoc/MethodAnnotationReturnTypeClassDoesNotExistRule.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\PhpDoc;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InClassNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use function class_exists;
+use function sprintf;
+
+/**
+ * @implements Rule<InClassNode>
+ */
+class MethodAnnotationReturnTypeClassDoesNotExistRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return InClassNode::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (!$scope->isInClass()) {
+			return [];
+		}
+		$classReflection = $scope->getClassReflection();
+		$errors = [];
+		foreach ($classReflection->getMethodTags() as $methodName => $methodTag) {
+			foreach ($methodTag->getReturnType()->getObjectClassNames() as $className) {
+				if (class_exists($className)) {
+					continue;
+				}
+
+				$errors[] = RuleErrorBuilder::message(
+					sprintf(
+						'PHPDoc tag @method %s() has invalid return type "%s", class does not exist',
```

Comment:
There are many existing rules you can draw inspiration from. They have one thing in common - they all call ClassCaseSensitivityCheck.

The one that's probably the closest is MixinRule. The ExistingClassesInPropertiesRule is also pretty close. That's gonna give you some ideas how to implement this.

Also, some more points:

1) Your rule isn't registered for any level.
2) You need to check return type, and parameter types too.
3) There should be similar rules for `@property` and also type aliases.

The last point can be done in a separate PR.

Reply:
1. What would be a good level for that check?
2. You mean `@return` and `@param`, right?

I pushed some changes after looking at the examples. Sorry if something is off, I'm new to the codebase

Reply:
You're checking only `$methodTag->getReturnType()`, you need to check `$methodTag->getParameters()` too.

Some things are still missing in your rule, like UnresolvableTypeHelper, and checking that the referenced class isn't a trait. As I said, please inspire yourself by the existing rules more :)

As for the level: level 2 would be great. But the rule needs to be registered only for bleedingEdge. Inspire yourself by the existing `conditionalTags` sections :) Thanks.